concepts/number-types/introduction.md

```diff
@@ -0,0 +1,126 @@
+| `Int8`   | 8           | -128 to 127                                                                                                 |
+| `Int16`  | 16          | -32_768 to 32_767                                                                                           |
+| `Int32`  | 32          | -2_147_483_648 to 2_147_483_647                                                                             |
+| `Int64`  | 64          | -9_223_372_036_854_775_808 to 9_223_372_036_854_775_807                                                     |
+| `Int128` | 128         | -170_141_183_460_469_231_731_687_303_715_884_105_728 to 170_141_183_460_469_231_731_687_303_715_884_105_727 |
+
+The smaller types use less memory and are faster than the larger types but have a smaller range of possible values.
+Consequently, the larger types use more memory and are slower than the smaller types, but have a larger range of possible values.
+
+Where you know that values will be within a certain range, it is best to use the smallest type possible to save memory and improve performance.
+
+The default integer type is `Int32`.
+To declare an integer with a specific type you can use the type name as a suffix, by adding `<number>_i<bit>`.
+
+```crystal
+1_i8.class
+# => Int8
+
+3.class
+# => Int32
+
+5_i128.class
+# => Int128
+```
+
+To convert between different integer types you can use the `to_i<bit>` method.
+The `to_i` method converts to the default integer type, which is `Int32`.
+
+```crystal
+1_i8.to_i16.class
+# => Int16
+
+2_i16.to_i.class
+# => Int32
+```
+
+## Unsigned integers types
+
+These the [unsigned integer][uint] types in Crystal:
+
+| Type      | Size (bits) | Range                                                    |
+| --------- | ----------- | -------------------------------------------------------- |
+| `UInt8`   | 8           | 0 to 255                                                 |
+| `UInt16`  | 16          | 0 to 65_535                                              |
+| `UInt32`  | 32          | 0 to 4_294_967_295                                       |
+| `UInt64`  | 64          | 0 to 18_446_744_073_709_551_615                          |
+| `UInt128` | 128         | 0 to 340_282_366_920_938_463_463_374_607_431_768_211_455 |
+
+The only difference to signed integers is that unsigned integers can only be positive.
+
+In situations where you only need to store positive values, unsigned integers are more efficient memory-wise (e.g. an 8 bit integer as twice the 0...255 rather than 0..127).
+A common use of positive-only integers is when working with indexes in an array.
+
+The default unsigned integer type is `UInt32`.
+To declare an unsigned integer with a specific type you can use the type name as a suffix, by adding `<number>_u<bit>`
+
+```crystal
+1_u8.class
+# => UInt8
+
+5_u128.class
+# => UInt128
+```
```

Comment:
all this can be removed

Reply:
I keept line 71, "To declare an unsigned integer with a specific type you can use the type name as a suffix, by adding `<number>_u<bit>`.", but I agree the rest I think can be removed.